nntrainer/tensor/cache_pool.cpp

```diff
@@ -11,51 +11,81 @@
  *
  */
 
-#include "memory_pool.h"
+#include "cache_pool.h"
+
 #include <limits>
 #include <numeric>
 #include <stdexcept>
 #include <vector>
 
-#include <cache_pool.h>
 #include <nntrainer_error.h>
 #include <nntrainer_log.h>
 #include <profiler.h>
 
 namespace nntrainer {
 
-void CacheElem::swapIn() {
-  std::lock_guard<std::mutex> lock(device_mutex);
-  void *buf = device->getBuffer(offset, length);
-  mem_data->setAddr((float *)buf);
-  mem_data->setValid(true);
-  active = true;
+namespace {
+
+inline const CachePolicy
+convertTensorLifespanToCachePolicy(const TensorLifespan lifespan) {
+  CachePolicy policy;
+
+  switch (lifespan) {
+  case TensorLifespan::UNMANAGED:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  case TensorLifespan::FORWARD_FUNC_LIFESPAN:
+    policy = CachePolicy::ALWAYS_SYNCED;
+    break;
+  case TensorLifespan::CALC_DERIV_LIFESPAN:
+    policy = CachePolicy::TEMPORAL;
+    break;
+  case TensorLifespan::CALC_GRAD_LIFESPAN:
+    policy = CachePolicy::TEMPORAL;
+    break;
+  case TensorLifespan::CALC_GRAD_DERIV_LIFESPAN:
+    policy = CachePolicy::TEMPORAL;
+    break;
+  case TensorLifespan::FORWARD_GRAD_LIFESPAN:
+    policy = CachePolicy::ITERATION_CONSIST;
+    break;
+  case TensorLifespan::FORWARD_DERIV_LIFESPAN:
```

Comment:
This assumes ```calcDerivatives"` includes ```Apply Gradient``` ?

Reply:
apply gradient is not implement yet in this PR.

Reply:
Yes. I see. I just ask to confirm.